### PR TITLE
Clean up squad sentence teachers

### PR DESCRIPTION
**Pull request by emilydinan**
Delete some unnecessary squad sentence teachers. Now there are two "sentence teachers: `squad:sentence` and `squad:fulldocsentence`. 

Use the flag `--include-context` to have the data formatted as `'text': <context>\n<question>` (if `True`) or `'context': <context>, 'text': <question>` (if `False`, this is the default behavior)

### CODE REVIEW

File: parlai/tasks/squad/agents.py [hunk 1]
```diff
@@ -383,15 +304,40 @@ class FulldocsentenceTeacher(FulldocTeacher):
...
+                           'separate field')
 
     def get(self, episode_idx, entry_idx=None):
+        action = {}
         episode = self.episodes[episode_idx][entry_idx]
-        entry = {'episode_done': episode['episode_done']}
-        context = ' '.join(episode['text'].split('\n')[:-1]).replace('\xa0', ' ')
+        context = ' '.join(
+            episode['text'].split('\n')[:-1]
+        ).replace('\xa0', ' ')
```
Comment (stephenroller) [inline parlai/tasks/squad/agents.py]: what's up with the weird character replacement?

Comment (emilydinan) [reply parlai/tasks/squad/agents.py]: great question...this is a non breaking space character and they were left in the text. in some cases, the correct answer span in the text contains this character and so if we don't remove it, the model has no possibility of getting the answer correct (if it returns an answer span)

Comment (stephenroller) [reply parlai/tasks/squad/agents.py]: Makes sense. It'd be nice if we had a comment here briefly explaining that.

Comment (emilydinan) [reply parlai/tasks/squad/agents.py]: added with commit ff68e3

**Pull request merged**